Add auto close bot

cjappl opened the pull request:
Alright, this is what we actually want.

Please add this bot to the repo. 
https://probot.github.io/apps/close-issue/

You can remove all other bots I requested on #702 

Here is an example of what will happen:
https://github.com/cjappl/defund12.org/issues/6
https://github.com/cjappl/defund12.org/issues/5

The bot will look for ALL OF:
  - "Steps to reproduce:"
  - "Expected behavior:"
  - "Current behavior:"
  - "Steps to reproduce"
  - "Detailed Description"

or ALL OF:
  - "defund"
  - "I am a resident of"
  - "YOUR NAME"
  - "YOUR EMAIL"
  - "YOUR PHONE NUMBER"
  - "YOUR ADDRESS"


If request has neither set, close automatically. Of course this is configurable. Let me know if you want it more or less strict!

@bingbongle @skullface @shortcircuit3

bingbongle commented:
Hey, this is definitely a step in the right direction!!!

Coupling together bug issues and email template requests seems dangerous. Can we not create separate configurations for both templates based on the labels that are automatically applied? That way we can change one bot without worrying about breaking the other. If we can solve this, we're absolutely golden 👍  Thanks for responding to feedback on this issue so quickly - we need this ASAP!

bingbongle commented:
@cjappl @shortcircuit3 @skullface

cjappl commented:
Hmm I see your point but I don't think there is anything we can do about it
in the very short term as the bot is set up.

I think if we keep the configurations specific enough, we won’t have a
problem.

As cited in my earlier comment, it is very unlikely that an email request
has the bug words, and very unlikely that a bug has the email words. Even
if they did, this takes care of 90% of the cases

As long as we stay as specific as we can without closing good requests I
think we will be golden!

On Mon, Jun 8, 2020 at 5:08 AM bing bong <notifications@github.com> wrote:

> @cjappl <https://github.com/cjappl> @shortcircuit3
> <https://github.com/shortcircuit3> @skullface
> <https://github.com/skullface>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/defund12/defund12.org/pull/781#issuecomment-640563049>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMDXY3C4EKWF32D6RFCVTLRVTID3ANCNFSM4NX2S5DQ>
> .
>

cjappl commented:
Going to update this PR in a bit to simplify the filter.

DO NOT MERGE AS IS

On Mon, Jun 8, 2020 at 7:53 AM Christopher Apple <
christopher.j.apple@gmail.com> wrote:

> Hmm I see your point but I don't think there is anything we can do about
> it in the very short term as the bot is set up.
>
> I think if we keep the configurations specific enough, we won’t have a
> problem.
>
> As cited in my earlier comment, it is very unlikely that an email request
> has the bug words, and very unlikely that a bug has the email words. Even
> if they did, this takes care of 90% of the cases
>
> As long as we stay as specific as we can without closing good requests I
> think we will be golden!
>
> On Mon, Jun 8, 2020 at 5:08 AM bing bong <notifications@github.com> wrote:
>
>> @cjappl <https://github.com/cjappl> @shortcircuit3
>> <https://github.com/shortcircuit3> @skullface
>> <https://github.com/skullface>
>>
>> —
>> You are receiving this because you were mentioned.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/defund12/defund12.org/pull/781#issuecomment-640563049>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADMDXY3C4EKWF32D6RFCVTLRVTID3ANCNFSM4NX2S5DQ>
>> .
>>
>

cjappl commented:
Yep closes and writes the comment to explain why!

On Mon, Jun 8, 2020 at 3:05 PM Roshan <notifications@github.com> wrote:

> *@rpalakkal* commented on this pull request.
>
> looks good - so this only closes empty requests that don't provide the
> required information?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/defund12/defund12.org/pull/781#pullrequestreview-426654521>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMDXY6PCN4QBWTEEVZLF53RVVOCTANCNFSM4NX2S5DQ>
> .
>

cjappl commented:
@bingbongle still waiting on your approval on this, or alternatively your "no go" and move to a custom solution!

To summarize the discussion in #issue-triage:

- We discussed moving to a much more lenient filter here, only looking for an instance of NAME, Feature, or description, which are all supplied as part of the templates
- If the issue doesn't have one of these, it is automatically closed, and the user is told explicitly what we look for with our bot 
- The overhead of doing a custom bot is non-negligible, so we wanted to see what you thought about this solution before diving into a custom solution. There does not seem to be an out-of-the-box solution available beyond this

NickCrews commented on .github/issue-close-app.yml:
```diff
@@ -0,0 +1,30 @@
+# The bot will only add a label if this property is set.
+#label: "closed by bot"
+
+# The bot will ignore any issues that are opened or reopened by the user names in exception
```
Remove these noisy comments please

NickCrews commented on .github/issue-close-app.yml:
```diff
@@ -0,0 +1,30 @@
+# The issue is judged to be legal if it includes all keywords from any of these two configs.
+# Or it will be closed by the app
+
+# Optional configuration:
```
Don't need this comment either

NickCrews commented on .github/ISSUE_TEMPLATE/REPORT_BUG.md:
```diff
@@ -6,10 +6,21 @@ labels: bug
 assignees: ''
 ---
 
+<!--Do not delete the lines in this template, or else your issue may be closed automatically!-->
```
Maybe phrase as "You must fill in the info within each section of the template, or a bot will close the issue automatically."

Just the phrasing "deleting lines" is a bit weird to me.

NickCrews commented on .github/issue-close-app.yml:
```diff
@@ -0,0 +1,30 @@
+# default value is false, which means keywords are case-sensitive
+caseInsensitive: true
+
+# the label that will be added when the bot close an issue
```
I would say delete these lines if we don't use the label feature.

NickCrews commented on .github/issue-close-app.yml:
```diff
@@ -0,0 +1,30 @@
+- content:
+# feature 
+  - "Feature"
+# The issue is judged to be legal if it includes all keywords from any of these two configs.
```
just leave out the "two". It's wrong now and it might just become more wrong later.

NickCrews commented on .github/issue-close-app.yml:
```diff
@@ -0,0 +1,30 @@
+# Comment that will be sent if an issue is judged to be closed.
```
I would delete this comment, and instead explain what the large-scale goal of this file is, because if I came across this later I would have no idea. Perhaps:
`# Config for a bot that automatically closes issues if they do not conform to some simple rules`

Thoughts?

cjappl commented:
@NickCrews  addressed your comments! Let me know what you think. I do think it's valuable to leave comments in that file that describe some of the behavior we aren't using right now.

Would you describe how to squash commits? I haven't used that on github before, in gitlab you just check a checkbox and it does it for you! 

Thanks for the help, really appreciate
